docs: Add docs for unmanaging modules (#1936)
kyma-project/lifecycle-manager: docs/contributor/resources/01-kyma.md

> **CAUTION:**
> Module referencing using NamespacedName and FQDN (Fully Qualified Domain Name) has been deprecated.

### **.spec.modules[].managed**

> The **managed** field determines whether Lifecycle Manager manages the module or not. By default it is set to `true`. By setting it to `false`, the module is opted out of management by Lifecycle Manager. This means that the module and all its related resources remain in the remote cluster in their current state at the time of unmanaging the module, but Lifecycle Manager does NOT reconcile those any longer.

Reviewer: ```suggestion
The **managed** field determines whether or not the Lifecycle Manager manages the module. By default, it is set to `true`. By setting it to `false`, the module is opted out of management by Lifecycle Manager. This means that the module and all its related resources remain in the remote cluster in the same state they were in when the module became unmanaged. From this time on, Lifecycle Manager does NOT reconcile the module and its resources.

> The **managed** field determines whether or not the Lifecycle Manager manages the module. By default, it is set to `true`. By setting it to `false`, the module is opted out of management by Lifecycle Manager. This means that the module and all its related resources remain in the remote cluster in the same state they were in when the module became unmanaged. From this time on, Lifecycle Manager does NOT reconcile the module and its resources.

Author: ```suggestion
The **managed** field determines whether or not the Lifecycle Manager manages the module. By default, it is set to `true`. By setting it to `false`, the module is opted out of management by Lifecycle Manager. This means that the module and all its related resources remain in the remote cluster in the same state they were in when the module became unmanaged. From this time on, Lifecycle Manager does NOT reconcile the module and its resources. In addition, the labels `operator.kyma-project.io/managed-by=kyma` and `operator.kyma-project.io/watched-by=kyma` are removed from the module's resources. This may be relevant if you for instance use those labels as filters for setting up telemetry auto-instrumentation.
```

Author: As quickly discussed with Icke.

> Once a module has been successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**.

Reviewer 2: I think you need to explicitly mention that also by removing the module entry when unmanaged, the module resources on the remote cluster will not be removed

> Once a module has been successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**.

Reviewer: ```suggestion
Once a module was successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**.
```

> If the **.spec.modules[].managed** field is set to `true` again, Lifecycle Manager picks up the management of the module again. Existing module resources in the remote cluster are left in tact but may be overwritten if the desired state has changed in the meantime, e.g., if the version within the used channel has progressed in the meantime.

Reviewer: ```suggestion
If the **.spec.modules[].managed** field is set back to `true`, Lifecycle Manager starts the module management again. The existing module resources in the remote cluster are left intact but may be overwritten if the desired state has changed in the meantime, for example, if the module's version within the used channel was updated.
```

> Once a module was successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**. As the module is already unmanaged, this still leaves the module and its related resources in the remote cluster.

Author: I think the last sentence ("even thogh") isn't what we want to say. We want to say that if the module is already unmanaged, the entry may also be removed entirely. Maybe something like:

```suggestion
To verify that a module was successfully unmanaged, check that the field **.status.modules[].state** has the status `Unmanaged`. Once the state is `Unmanaged`, the module's entry can be deleted from **.spec.modules[]**. This will only remove the module's entry in the KymaCR. The module and its related resources remain in the remote cluster.
```

> Once a module was successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**. As the module is already unmanaged, this still leaves the module and its related resources in the remote cluster.

Reviewer: ```suggestion
To verify that a module was successfully unmanaged, check that the field **.status.modules[].state** has the status `Unmanaged`. Once the state is `Unmanaged`, you can delete the module's entry from **.spec.modules[]** in the Kyma CR.  Nevertheless, the module and its related resources remain in the remote cluster.
```
Ok, how about this version?

> Once a module was successfully unmanaged, the module's status in **.status.modules[].state** shows `Unmanaged`. Once the `Unmanaged` state shows, the module's entry can also be deleted entirely from **.spec.modules[]**. As the module is already unmanaged, this still leaves the module and its related resources in the remote cluster.

Author: Thanks! LIke the suggestion. Deleting is optional and it actually doesn't make a difference if the unmanaged module is deleted from the KymaCR or not. It is just a cleanup possibility to only list things in KymaCR that are managed by us.

> 

Author: > **CAUTION:**
> By unmanaging the module, you explicitly opt out of any support for the module.

Author: Checked with Jan, this is not needed for now.

> **CAUTION:**

> > When switching between values of **.spec.modules[].managed**, the user MUST wait for the new state to be reflected in **.status.modules[].state** before removing the module's entry from **.status.modules[]**. If the entry is removed before the current state is reflected properly in **.status.modules[].state**, it may lead to unpredictable behavior that is hard to be recovered from.

Reviewer: ```suggestion
> When you switch values of **.spec.modules[].managed**, you MUST wait for the new state to be reflected in **.status.modules[].state** before you remove the module's entry from **.status.modules[]**. If the entry is removed before the current state is reflected properly in **.status.modules[].state**, it may lead to unpredictable behavior that is hard to recover from.
```
